invisible_cities/io/dst_io.py

```diff
@@ -58,9 +58,12 @@ def _store_pandas_as_tables(h5out : tb.file.File, df : pd.DataFrame, group_name
     table = getattr(group, table_name)
     if not np.array_equal(df.columns, table.colnames):
         raise TableMismatch(f'dataframe differs from already existing table structure')
-    for indx in df.index:
-        tablerow = table.row
-        for colname in table.colnames:
-            tablerow[colname] = df.at[indx, colname]
-        tablerow.append()
+
+    arr = df.to_records(index=False)
+    #hack to transform object numpy dtype to string
+    data_types = [(col, arr[col].dtype if arr[col].dtype != 'O' else \
+                   f'S{str_col_length}') for col in arr.dtype.names]
```

Comment:
It's a bit hard to read. Can we define a function (even inline) to do that?

Reply:
How about now?